persistence/azure-table/transactions_sharing_astp_[5,).partial.md

```diff
@@ -0,0 +1,23 @@
+## Sharing the transaction
+
+Once a behavior is introduced to identify the partition key for a given message, it is possible to share a batch, represented in Azure Table by `List<TableTransactionAction>` between both the Saga persistence and business data. This list can then be used to persist document updates for both concerns atomically.
+
+### Within a handler method using `IMessageHandlerContext`
+
+To use the shared `List<TableTransactionAction>` in a message handler:
+
+snippet: HandlerSharedTransaction
+
+#### Testing
+
+The `TestableAzureTableStorageSession` class in the `NServiceBus.Testing` namespace has been provided to facilitate [testing a handler](/nservicebus/testing/) that utilizes the shared transaction feature.
+
+### With dependency injection
+
+For custom types that require access to the shared `List<TableTransactionAction>`:
+
+snippet: TransactionalBatchRegisteredWithDependencyInjectionResolvedInCustomType
+
+And alternatively to using the the extension method `IMessageHandlerContext.SynchronizedStorageSession.GetSharedTransactionalBatch()`:
```

Comment:
I think something is off here and in the previous documentation. This method does not exist and should probably be https://github.com/Particular/NServiceBus.Persistence.AzureTable/blob/master/src/NServiceBus.Persistence.AzureTable/SynchronizedStorage/SynchronizedStorageSessionExtensions.cs#L13

Reply:
Pushed